Docs updates for Umbraco Forms 10.2. (#4281)
umbraco/UmbracoDocs: Add-ons/UmbracoForms/Developer/Configuration/index.md

If you have created a custom template and would like to use that as the default instead, you can set the path here using this configuration setting.

### RemoveProvidedEmailTemplate

> From Forms 10.2, if you have created email templates to use with the "send Razor email" workflow, the provided template can be removed from selection. To do this, set this value to `true`.

Reviewer: ```suggestion
From Forms 10.2, the provided template can be removed from the selection if you have created email templates for the "send Razor email" workflow. To do this, set this value to `true`.
```

### RemoveProvidedFormsTemplates
Similarly, from Forms 10.2, the provided form templates available from the form creation dialog can be removed from selection. To do this, set this configuration value to `true`.

### FormElementHtmlIdPrefix
By default the value of HTML `id` attribute rendered for fieldsets and fields using the default theme is the GUID associated with the form element. Although [this is valid](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id), some browsers, particularly Safari, may report issues with this if the identifier begins with a number. To avoid such issues, from Forms 10.2, the attribute values can be prefixed with the value provided in this configuration element.

For example, providing a value of `"f_"` will apply a prefix of "f_" to each fieldset and field `id` attribute.

> ###

Reviewer 2: Is there a heading missing here?

> Forms 10.2 introduced the ability to configure settings for the field, workflow, datasource and prevalue sources.  The default behaviour for Forms when a new field or workflow is added to to a form is for each setting to be empty. The values are then completed in by the editor.  All settings defined on the type are displayed for entry.

Reviewer 2: ```suggestion
Forms 10.2 introduced the ability to configure settings for the field, workflow, data source, and prevalue sources.  The default form behavior, when a new field or workflow is added to a form, is for each setting to be empty. The values are then completed by the editor.  All settings defined on the type are displayed for entry.
```

It consists of four dictionaries, one for each type:

- `DataSourceTypes`
- `FieldTypes`
- `PrevalueSourceTypes`
- `WorkflowTypes`

Each dictionary can be identified using the GUID or alias of the type as the key. The value is set to the following structure that contains three settings:

```json
{
  "IsHidden": true|false,
  "DefaultValue": "",
  "IsReadOnly": true|false
}
```

- `IsHidden` - if provided and set to true the setting will be hidden and will always have an empty value.
- `DefaultValue` - if provided the value will be pre-filled when a type using it is created.
- `IsReadOnly` - used in conjunction with the above, if set the field won't be editable and hence whatever is set as the `DefaultValue` won't be able to be changed. If set to false (or omitted) the editor can change the value from the default.

In this example, the sender address field on a workflow for sending emails can be hidden, such that the system configured value is always used:

```json
  "SettingsCustomization": {
    "WorkflowTypes": {
      "sendEmailWithRazorTemplate": {
        "SenderEmail": {
          "IsHidden": true
        }
      }
    },
  }
```

> Here an organisation approved reCAPTCHA score threshold is defined, that can't be changed by editors:

Reviewer 2: ```suggestion
Here an organization-approved reCAPTCHA score threshold is defined, that can't be changed by editors:
```

```json
  "SettingsCustomization": {
    "FieldTypes": {
      "recaptcha3": {
        "ScoreThreshold": {
          "DefaultValue": "0.8",
          "IsReadOnly": true
        }
      }
    },
  }
```

In order to configure this setting, you will need to know the GUID or alias for the type and the property name for each setting. You can find [these values for the built-in Forms types here](./type-details.md).

Take care to not hide any settings that are required for the particular field or workflow type (for example, the `Subject` field for email workflows). If you do that, the item will fail validation when an editor tries to create it.

> The default value and read-only settings apply for most setting types, with the exception of complex ones where a default string value isn't appropriate. An example of one of these is the field mapper used in the "Send to URL" workflow.

Reviewer: ```suggestion
The default value and read-only settings apply to most setting types. There is an exception for complex ones where a default string value isn't appropriate. An example of one of these is the field mapper used in the "Send to URL" workflow.
```
